### ISSUE TITLE
décrire les différentes opérations dans l'API

### ISSUE THREAD

**Issue by evaletolab** [karibou-ch/karibou-quad#12]
Chaque plot est basé sur une opération, et une série de données. 
- [ ] décrire le but de se plot (créer une nouvelle classe)
- [ ] proposer un set de méthodes (opérations, transformation, mapping)
- [ ] proposer les interfaces pour Input, et Output
- [ ] découpler le service de Mongodb

**Comment by jcavat** [comment]
Les composants 'chart' sont génériques. On leur passe les x et les y dans un tableau. La aussi, pour l'instant il s'agit d'expérimentation. A voir ce qui sera réellement intéressant comme chart.

**Comment by evaletolab** [comment]
ok cf #16

**Comment by evaletolab** [comment]
Une fois que tu auras une idée plus précise, tu pourras créer les classes suivantes:
class `CustomerInsight{}`, 
class `VendorInsight{}` 
class `OrdersInsight{}`
et peut être la classe de data provider `MongodbProvider{}`
Tu en penses quoi?

**Comment by jcavat** [comment]
Oui, à terme je vais décrire le format de réponse du serveur REST pour qu'on puisse migrer vers une autre solution de mapping en gardant les mêmes routes. Je proposerai des classes côté client. Pour le provider, il s'agit d'un provider qui interroge un serveur REST sans savoir quel type de base de données se cache derrière. Du coup, le nom `database.service.ts` me convient mieux.

**Comment by evaletolab** [comment]
je ne comprends pas pourquoi tu as besoin d'une isolation REST entre le dataprovider/consumer? Il suffit d'exposer des méthodes d'un service qui utilise mongodb.

**Comment by jcavat** [comment]
C'est une manière de faire que d'isoler un service derrière une api rest. J'aurais pu faire autrement. Ce sera un point pour les améliorations futures.
